apple/internal/partials/framework_import.bzl

```diff
@@ -69,28 +79,54 @@ def _framework_import_partial_impl(ctx, targets, targets_to_avoid):
         if framework_relative_dir:
             parent_dir = paths.join(parent_dir, framework_relative_dir)
 
-        bundle_files.append(
-            (processor.location.framework, parent_dir, depset([file])),
-        )
+        # check to see if the the parent is "Foo.[extension]" and the file is "Foo", thus "Foo.framework/Foo", so the binary within the framework.
+        if paths.replace_extension(parent_dir, "") == file.basename:
+            stripped = intermediates.file(
+                ctx.actions,
+                ctx.label.name,
+                paths.join("_imported_frameworks", file.basename),
+            )
+            bundle_files.append(
+                (processor.location.framework, parent_dir, depset([stripped])),
+            )
+
+            args = slicer_args + ["--in", file.path, "--out", stripped.path]
+            all_binaries = extra_binaries + [main_binary]
+            for binary in all_binaries:
+                args.append(binary.path)
+
+            ctx.actions.run(
+                inputs = [file] + all_binaries,
+                tools = [ctx.executable._realpath],
+                executable = ctx.executable._dynamic_framework_slicer,
+                outputs = [stripped],
+                arguments = args,
+                mnemonic = "DynamicFrameworkSlicer",
+            )
+        else:
+            bundle_files.append(
+                (processor.location.framework, parent_dir, depset([file])),
+            )
 
     return struct(bundle_files = bundle_files)
 
-def framework_import_partial(targets, targets_to_avoid = []):
+def framework_import_partial(targets, targets_to_avoid = [], extra_binaries = []):
     """Constructor for the framework import file processing partial.
 
     This partial propagates framework import file bundle locations. The files are collected through
     the framework_import_aspect aspect.
-
```

Comment:
You removed this line which is likely what trips things up, the spec is a blank line before the "Args:" block.

Reply:
Not even sure how I managed that...!